include/maths/CLocalOutlierFactors.h

```diff
@@ -94,12 +106,18 @@ class MATHS_EXPORT CLocalOutlierFactors {
     //! \param[in] points The points for which to compute scores.
     //! \param[out] scores The scores of \p points.
     template<typename POINT>
-    static void normalizedLdof(std::size_t k,
-                               bool project,
-                               const std::vector<POINT>& points,
-                               TDoubleVec& scores) {
-        CLdof<POINT, TKdTree<POINT>> ldof(k);
-        compute(ldof, project, points, scores);
+    static void
+    normalizedLdof(std::size_t k, bool project, std::vector<POINT> points, TDoubleVec& scores) {
```

Comment:
These all look the same except for the types that are in {CLof, CLdof, CDistancekNN, etc.} Good candidates for templating? Just a suggestion.